api/envoy/extensions/filters/http/tap/v3/tap.proto

```diff
@@ -26,4 +26,21 @@ message Tap {
   // Common configuration for the HTTP tap filter.
   common.tap.v3.CommonExtensionConfig common_config = 1
       [(validate.rules).message = {required: true}];
+
+  // Specific configuration for the HTTP tap filter
+  HttpSpecificConfig http_specific_config = 2;
+}
+
+// Specific configuration for the HTTP tap filter
+message HttpSpecificConfig {
+  // output configuration
+  OutputConfig output_config = 1;
+}
+
+// Configuration to control trace output
+message OutputConfig {
+  // Indicates whether HTTP taps filter produce the time stamp for request/response message
+  // Request time stamp is stored after recv request header
+  // Response time stamp is stored after recv response header
+  bool record_req_resp_msg_caught_time = 1;
```

Comment:
I will take the first comments. that sounds good.
for second one, I wan to use one flag. as you know, from our customer, if want to timestamps, it will use both.